Make GitHub Your Code Repository (#8)
RhoInc/blog: _posts/2019-05-01-Make-GitHub-Your-Code-Repository.md

---
layout: post
title: Make GitHub Your Code Repository

> ---

Reviewer: ```suggestion
author: Spencer Childress
---
```

> Downloading code from [GitHub](https://github.com/)� manually is straightforward: navigate to the repository website, download the ZIP file, and extract it to your working directory.

Reviewer: __
```suggestion
_This post shares information from our presentation at this year's [PhUSE Connect conference](https://www.phuse.eu/us-connect19).  The [paper](https://rhoinc.github.io/publication-library/pubs/PhUSE2019_Childress.pdf) and [presentation](https://rhoinc.github.io/publication-library/pubs/PhUSE2019_Childress_slides.pdf) from this presentation and others are available in our [Publications Library](https://rhoinc.github.io/publication-library/)_

Downloading code from [GitHub](https://github.com/)® manually is straightforward: navigate to the repository website, download the ZIP file, and extract it to your working directory.
```

> Downloading code from [GitHub](https://github.com/)� manually is straightforward: navigate to the repository website, download the ZIP file, and extract it to your working directory.

Reviewer: jekyll was complaining about the copyright symbol locally, but might be ok on pages ... just double check once we merge.

However, because this process is manual it needs to be repeated whenever the repository changes, such as when the developer applies bug fixes or incorporates new features.
SAS� and R provide users the tools to programmatically download and source repositories housed on GitHub.

SAS and R both have a base set of functionality, but they differ in that R provides access to user-created packages, code bundles that extend R's functionality, hosted on services like the [Comprehensive R Archive Network](https://cran.r-project.org/) (CRAN) and GitHub.
CRAN hosts established, vetted packages while GitHub tends to host more developmental packages.

> GitHub is not limited to R code, however, so the macro described in this paper extends to SAS access to remote code repositories so ingrained in R development.

Reviewer: ```suggestion
GitHub is not limited to R code, however, so the macro described here extends to SAS access to remote code repositories so ingrained in R development.
```

## Sourcing Code from GitHub

> In R, the [devtools package](https://cran.r-project.org/web/packages/devtools/index.html) contains a function named `install_github` that allows users to automate the installation of packages directly from GitHub:

Reviewer: devtools is really just pulling`install_github` from `remotes`, so probably better to reference that. 

```suggestion
In R, the [remotes package](https://cran.r-project.org/web/packages/remotes/index.html) contains a function named `install_github` that allows users to automate the installation of packages directly from GitHub:
```

```R
# Install and source the devtools package.

> install.packages("devtools") 

Reviewer: ```suggestion
install.packages("remotes") 
```

> library(devtools) 

Reviewer: ```suggestion
library(remotes) 
```
